processing/src/main/java/io/druid/segment/data/DeltaLongEncodingWriter.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+public class DeltaLongEncodingWriter implements CompressionFactory.LongEncodingWriter
+{
+
+  private final long base;
+  private VSizeLongSerde.LongSerializer serializer;
+  private int bitsPerValue;
+
+  public DeltaLongEncodingWriter(long base, long delta)
+  {
+    this.base = base;
+    this.bitsPerValue = VSizeLongSerde.getBitsForMax(delta + 1);
```

Comment:
getBitsForMax(n) returns bits required to encode at most n unique values. delta is the difference between the smallest and largest value, which means there would be delta + 1 unique values since smallest and largest value are both included in the set. 

For the delta = Long.MaxValue case it would cause an exception, and I can change the strategy to select Longs format for this case. Also going to add a test with exactly max long delta.